File: dart-jaguar/lib/model/emoji_reaction_base_all_of_user.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';


part 'emoji_reaction_base_all_of_user.jser.dart';

class EmojiReactionBaseAllOfUser {
   /* ID of the user.  */
  @Alias('id', isNullable: false,  )
  final int id;
   /* Email of the user.  */
  @Alias('email', isNullable: false,  )
  final String email;
   /* Full name of the user.  */
  @Alias('full_name', isNullable: false,  )
  final String fullName;
   /* Whether the user is a mirror dummy.  */
  @Alias('is_mirror_dummy', isNullable: false,  )
  final bool isMirrorDummy;
  

  EmojiReactionBaseAllOfUser(
      

{
     this.id = null,  
     this.email = null,  
     this.fullName = null,  
     this.isMirrorDummy = null 
    
    }
  );

  @override
  String toString() {
    return 'EmojiReactionBaseAllOfUser[id=$id, email=$email, fullName=$fullName, isMirrorDummy=$isMirrorDummy, ]';
  }
}

@GenSerializer(nullableFields: true)
class EmojiReactionBaseAllOfUserSerializer extends Serializer<EmojiReactionBaseAllOfUser> with _$EmojiReactionBaseAllOfUserSerializer {

}

File: scala-akka-http-server/src/main/scala/org/openapitools/server/model/EmojiReactionBaseAllOfUser.scala
package org.openapitools.server.model


/**
 * Whether the user is a mirror dummy. Dictionary with data on the user who added the reaction, including the user ID as the `id` field.  **Note**: In the [events API](/api/get-events), this `user` dictionary confusing had the user ID in a field called `user_id` instead.  We recommend ignoring fields other than the user ID.  **Deprecated** and to be removed in a future release once core clients have migrated to use the `user_id` field. 
 *
 * @param id ID of the user.  for example: ''null''
 * @param email Email of the user.  for example: ''null''
 * @param fullName Full name of the user.  for example: ''null''
 * @param isMirrorDummy Whether the user is a mirror dummy.  for example: ''null''
*/
final case class EmojiReactionBaseAllOfUser (
  id: Option[Int],
  email: Option[String],
  fullName: Option[String],
  isMirrorDummy: Option[Boolean]
)

